ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -146,10 +147,17 @@ public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> stor
     return new StoreInfo(messageReadSet, messageInfos);
   }
 
-  public void downloadBlob(BlobId blobId, OutputStream outputStream) throws StoreException {
+  public void downloadBlob(CloudBlobMetadata cloudBlobMetadata, BlobId blobId, OutputStream outputStream) throws StoreException {
```

Comment:
Make this package private.  Unless this will be called outside the cloud package, in which case the caller probably shouldn't know about CloudBlobMetadata.
Also javadoc.